core/templates/dev/head/pages/story_editor/main_editor/StoryNodeEditorDirective.js

```diff
@@ -32,17 +32,53 @@ oppia.directive('storyNodeEditor', [
         '$scope', '$rootScope', '$uibModal', 'StoryEditorStateService',
         'StoryUpdateService', 'UndoRedoService', 'EVENT_STORY_INITIALIZED',
         'EVENT_STORY_REINITIALIZED', 'EVENT_VIEW_STORY_NODE_EDITOR',
+        'AlertsService',
         function(
             $scope, $rootScope, $uibModal, StoryEditorStateService,
             StoryUpdateService, UndoRedoService, EVENT_STORY_INITIALIZED,
-            EVENT_STORY_REINITIALIZED, EVENT_VIEW_STORY_NODE_EDITOR) {
+            EVENT_STORY_REINITIALIZED, EVENT_VIEW_STORY_NODE_EDITOR,
+            AlertsService) {
+          var _recalculateAvailableNodes = function() {
+            $scope.newNodeId = undefined;
+            $scope.availableNodes = [{
+              id: undefined,
```

Comment:
What's the difference between id=null and id=undefined? Probably should be null throughout, right?

That said, I think you shouldn't represent the null node as a node -- since it conceptually isn't one. It's going to cause problems if other devs use $scope.availableNodes for calculations, e.g. of how many nodes there are in the story. Better to have a separate field altogether for the null node and not lump it with the rest of the (real) nodes.

Reply:
The null is for the placeholder text 'Pick a chapter' and the `undefined` was for the option which is used to create a new chapter for the story.
Will separate out the available nodes list to have only the nodes and not id=null and undefined elements.

Reply:
Done, removed the use of null and undefined.